custom_components/mqtt_vacuum_camera/utils/vacuum/mqtt_vacuum_services.py

```
@@ -64,23 +90,44 @@
+        elif zone_lists and isinstance(zone_lists, list):
+            for sub_zone in zone_lists:
+                if len(sub_zone) != 4:
+                    got_error = "zone_list_length"
+        elif zone_lists and (not isinstance(zone_lists, list)):
+            got_error = "zone_must_be_list"
+        elif zone_ids and isinstance(zone_ids, list):
+            zone_lists = zone_ids
+        elif not isinstance(zone_ids, list):
+            got_error = "zoneid_must_be_list"
+
+        if got_error is not "No Errors":
```

Ruff (0.8.2)

Use `!=` to compare constant literals

Replace `is not` with `!=`

(F632)